src/View/Requirements.php

```diff
@@ -136,10 +136,13 @@ public static function javascript($file, $options = [])
      *
      * @param string     $script       The script content as a string (without enclosing `<script>` tag)
      * @param string|int $uniquenessID A unique ID that ensures a piece of code is only added once
+     * @param array $options List of options. Available options include:
+     * - 'type' : Specifies the type of script
+     * - 'crossorigin' : Cross-origin policy for the resource
      */
-    public static function customScript($script, $uniquenessID = null)
+    public static function customScript($script, $uniquenessID = null, $options = [])
```

Comment:
Unfortunately we can't add new arguments to existing public API in a minor release - you will need to find another way to add this, or else wait until we have a `6` branch to target a major release.

I'm going to close this PR - feel free to open a new one if you find another way to do this, or when we've created the `6` branch for targetting a major release (which is likely to be early next year).

Reply:
Would any of these be an applicable way to implement this?
- A new function to change the current options for further customScripts reqs.
- Detect if the template is surrounded in script tags, if true render the template as is.
- Replace uniquenessID with options, if set to a non array value treat as usual. Otherwise parse array that can contain uniquenessID as a key.

Reply:
> A new function to change the current options for further customScripts reqs.

This one could be dangerous, since you might want to only affect _one_ call to `customScript()`, but forget to set the value back to what it was beforehand, and end up affecting _all_ calls to `customScript`().

You could introduce a new `customScriptWithAttributes()` method though, which is just exactly what this PR was doing except as its own new method.

> Detect if the template is surrounded in script tags, if true render the template as is.

I'd need more information about what this means, but it sounds like it could be a breaking change depending on what the current behaviour in that scenario is.

> Replace uniquenessID with options, if set to a non array value treat as usual. Otherwise parse array that can contain uniquenessID as a key.

We could do this, except we can't change the name of the parameter (due to [PHP introducing support for named arguments in PHP 8.0](https://www.php.net/manual/en/functions.arguments.php#functions.named-arguments)) and it would be weird to have a parameter called `uniquenessID` that actually handles a lot of options. Because of that I wouldn't support this change.